core/domain/collection_domain.py

```diff
@@ -577,20 +838,39 @@ def _find_node(self, exploration_id):
     def get_node(self, exploration_id):
         """Retrieves a collection node from the collection based on an
         exploration ID.
+
+        Args:
+            exploration_id: str. The id of the exploration.
+
+        Returns:
+            CollectionNode. If the list of nodes contains the given exploration
+            then it will return the node, else None.
         """
         for node in self.nodes:
             if node.exploration_id == exploration_id:
                 return node
         return None
 
     def add_node(self, exploration_id):
+        """Adds a collection node to the collection based on an
```

Comment:
Hm, this is a bit tricky to word, but I think "based on an" is a bit vague. How about:

Adds a new node to the collection; the new node represents the given exploration_id.